File: data-prep/dataprep-model/src/main/scala/de/hpi/isg/dataprep/model/targets/Metadata.scala
package de.hpi.isg.dataprep.model.targets

import de.hpi.isg.dataprep.util.Nameable

/**
  * @author Lan Jiang
  * @since 2018/5/29
  */
abstract class Metadata extends Target with Nameable {

  var name : String

  override def toString: String = name
}

File: data-prep/dataprep-util/src/main/java/de/hpi/isg/dataprep/util/Metadata.java
package de.hpi.isg.dataprep.util;

/**
 * @author Lan Jiang
 * @since 2018/8/24
 */
public enum Metadata {
    FILE_ENCODING("file-encoding"),
    DELIMITER("delimiter"),
    ESCAPE_CHARACTERS("escape-characters"),
    QUOTE_CHARACTER("quote-character"),
    PROPERTY_DATATYPE("property-datatype");

    private String metadata;

    Metadata(String string) {
        this.metadata = string;
    }

    public String getMetadata() { return metadata;}
}
